presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -318,49 +383,58 @@ else if (!forBroker) {
             String orderByExpressions = topNColumnOrderingMap.entrySet().stream().map(entry -> selections.get(entry.getKey()).getDefinition() + (entry.getValue().isAscending() ? "" : " DESC")).collect(Collectors.joining(", "));
             query = query + " ORDER BY " + orderByExpressions;
         }
+        query = query + limitClause;
+        return query;
+    }
+
+    /**
+     * Convert the current context to a Pinot SQL
+     */
+    public PinotQueryGenerator.GeneratedPinotQuery toSqlQuery(PinotConfig pinotConfig, ConnectorSession session)
+    {
+        int nonAggregateShortQueryLimit = PinotSessionProperties.getNonAggregateLimitForBrokerQueries(session);
+        boolean isQueryShort = hasAggregation() || limit.orElse(Integer.MAX_VALUE) < nonAggregateShortQueryLimit;
+        boolean forBroker = !PinotSessionProperties.isForbidBrokerQueries(session) && isQueryShort;
+        if (!pinotConfig.isAllowMultipleAggregations() && aggregations > 1 && !groupByColumns.isEmpty()) {
```

Comment:
Should we still check for this in the SQL codepath ? This limitation really only stemmed from PQL.

Reply:
removed this.